plugins/woocommerce-blocks/assets/js/blocks/product-gallery/style.scss

```diff
@@ -326,3 +326,17 @@ body.wc-block-product-gallery-dialog-open {
 		}
 	}
 }
+
+// On mobile devices (<600px):
+// - Product gallery groups are stacked vertically
+// - Thumbnails appear below the main gallery image and are arranged horizontally in a row
+@include breakpoint("<600px") {
+	.wp-block-group.is-layout-flex {
```

Comment:
This is too generic as Tom pointed out and also if you give it some specificity like adding `wp-block-woocommerce-product-gallery`, you won't have to use `!important`.